File: src/main/java/pl/mjedynak/idea/plugins/pit/cli/PitCommandLineArgumentsContainerImpl.java
package pl.mjedynak.idea.plugins.pit.cli;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import pl.mjedynak.idea.plugins.pit.cli.model.PitCommandLineArgument;

public class PitCommandLineArgumentsContainerImpl implements PitCommandLineArgumentsContainer {

    private Map<PitCommandLineArgument, String> map = new ConcurrentHashMap<PitCommandLineArgument, java.lang.String>();

    @Override
    public void put(PitCommandLineArgument argument, String value) {
        map.put(argument, value);
    }

    @Override
    public String get(PitCommandLineArgument argument) {
        return map.get(argument);
    }
}

File: src/main/kotlin/pl/mjedynak/idea/plugins/pit/cli/PitCommandLineArgumentsContainerImpl.kt
package pl.mjedynak.idea.plugins.pit.cli

import pl.mjedynak.idea.plugins.pit.cli.model.PitCommandLineArgument
import java.util.concurrent.ConcurrentHashMap

class PitCommandLineArgumentsContainerImpl : PitCommandLineArgumentsContainer {
    private val map = ConcurrentHashMap<PitCommandLineArgument, String>()

    override fun put(
        argument: PitCommandLineArgument,
        value: String,
    ) {
        map[argument] = value
    }

    override fun get(argument: PitCommandLineArgument): String? = map[argument]
}
